{
  "description": "OIDC authentication without retry",
  "schemaVersion": "1.18",
  "runOnRequirements": [
    {
      "minServerVersion": "7.0",
      "auth": true,
      "authMechanism": "MONGODB-OIDC"
    }
  ],
  "createEntities": [
    {
      "client": {
        "id": "authClient"
      }
    },
    {
      "client": {
        "id": "client0",
        "uriOptions": {
          "authMechanism": "MONGODB-OIDC",
          "authMechanismProperties": {
            "$$placeholder": 1
          },
          "retryReads": true,
          "retryWrites": true
        },
        "observeEvents": [
          "commandStartedEvent",
          "commandSucceededEvent",
          "commandFailedEvent"
        ]
      }
    },
    {
      "database": {
        "id": "database0",
        "client": "client0",
        "databaseName": "test"
      }
    },
    {
      "collection": {
        "id": "collection0",
        "database": "database0",
        "collectionName": "collName"
      }
    }
  ],
  "initialData": [
    {
      "collectionName": "collName",
      "databaseName": "test",
      "documents": [

      ]
    }
  ],
  "tests": [
    {
      "description": "A simple find operation should succeed",
      "operations": [
        {
          "name": "find",
          "arguments": {
            "filter": {
            }
          },
          "object": "collection0",
          "expectResult": [

          ]
        }
      ],
      "expectEvents": [
        {
          "client": "client0",
          "events": [
            {
              "commandStartedEvent": {
                "command": {
                  "find": "collName",
                  "filter": {
                  }
                }
              }
            },
            {
              "commandSucceededEvent": {
                "commandName": "find"
              }
            }
          ]
        }
      ]
    },
    {
      "description": "Write command should reauthenticate when receive ReauthenticationRequired error code and retryWrites=true",
      "operations": [
        {
          "name": "failPoint",
          "object": "testRunner",
          "arguments": {
            "client": "client0",
            "failPoint": {
              "configureFailPoint": "failCommand",
              "mode": {
                "times": 1
              },
              "data": {
                "failCommands": [
                  "insert"
                ],
                "errorCode": 391
              }
            }
          }
        },
        {
          "name": "insertOne",
          "object": "collection0",
          "arguments": {
            "document": {
              "_id": 1,
              "x": 1
            }
          }
        }
      ],
      "expectEvents": [
        {
          "client": "client0",
          "events": [
            {
              "commandStartedEvent": {
                "command": {
                  "insert": "collName",
                  "documents": [
                    {
                      "_id": 1,
                      "x": 1
                    }
                  ]
                }
              }
            },
            {
              "commandFailedEvent": {
                "commandName": "insert"
              }
            },
            {
              "commandStartedEvent": {
                "command": {
                  "insert": "collName",
                  "documents": [
                    {
                      "_id": 1,
                      "x": 1
                    }
                  ]
                }
              }
            },
            {
              "commandSucceededEvent": {
                "commandName": "insert"
              }
            }
          ]
        }
      ]
    }
  ]
}
